TempleEight/docs/guide/enumeration.md

```diff
@@ -3,3 +3,214 @@ id: enumeration
+To do this, you can widen the readability of the service using the `#readable(by: all)` metadata.
+If we apply this to our previous Templefile example:
+
+
+```
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #authMethod(email);
+}
+
+ExampleUser: service {
+  name: string;
+
+  #auth;
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+
+  #readable(by: all);
+  #enumerable;
+}
+```
+
+Repeating the previous requests should result in all entities being returned when listing, using both access tokens:
+
+```bash
+# Deploy the project locally
+❯❯❯ source deploy.sh
+...
+
+# Create two access tokens, extract them from the JSON response, and assign each to an environment variable
+❯❯❯ TOKEN1=$(curl -X POST $KONG_ENTRY/api/auth/register -d '{"email": "temple1@temple.com", "password":"abcdefgh"}' | jq -r .AccessToken)
+❯❯❯ TOKEN2=$(curl -X POST $KONG_ENTRY/api/auth/register -d '{"email": "temple2@temple.com", "password":"abcdefgh"}' | jq -r .AccessToken)
+
+# Create two new entities in the example service, one for each token
+❯❯❯ curl -X POST $KONG_ENTRY/api/example-service -d '{"foo": "Hello", "bar": 123}' -H "Authorization: Bearer $TOKEN1"
+{"id":"596d62db-8587-11ea-9d77-0242ac160003","foo":"Hello","bar":123}
+❯❯❯ curl -X POST $KONG_ENTRY/api/example-service -d '{"foo": "World", "bar": 456}' -H "Authorization: Bearer $TOKEN2"
+{"id":"60e013cd-8587-11ea-9d77-0242ac160003","foo":"World","bar":456}
+
+# List all entities stored in the example service, using each token:
+❯❯❯ curl -X GET $KONG_ENTRY/api/example-service/all -H "Authorization: Bearer $TOKEN1"
+{
+  "ExampleServiceList":[
+    {"id":"596d62db-8587-11ea-9d77-0242ac160003","foo":"Hello","bar":123},
+    {"id":"60e013cd-8587-11ea-9d77-0242ac160003","foo":"World","bar":456}
+  ]
+}
+
+❯❯❯ curl -X GET $KONG_ENTRY/api/example-service/all -H "Authorization: Bearer $TOKEN2"
+{
+  "ExampleServiceList":[
+    {"id":"596d62db-8587-11ea-9d77-0242ac160003","foo":"Hello","bar":123},
+    {"id":"60e013cd-8587-11ea-9d77-0242ac160003","foo":"World","bar":456}
+  ]
+}
+```
+
+
+### Enumerating on `#auth` services
+:::important
+Services which contain the `#auth` metadata are special, in that they provide a 1-to-1 mapping between an access token and an entity within that service.
+This means that there can only be at most one entity for a given access token.
+
+By default any service with a `#auth` block are implicitly `#readable(by: this)`, allowing only the access token that created the entity to read the information stored.
+This means that it is not possible to have an `#enumerable` tag on an `#auth` service, since this would only ever return the list containing the entity for that access token.
+
+However, if the readability of the service is expanded to `#readable(by: all)`, usage of `#enumerable` is permitted, and will return all entities in that service.
```

Comment:
```suggestion
By default any service with a `#auth` block are implicitly `#readable(by: this)`, allowing only the access token that created the entity to read the information stored.
This means that without overriding to `readable(by: all)`, it is not possible to have an `#enumerable` tag on an `#auth` service, since this would only ever return the list containing the entity for that access token.
```